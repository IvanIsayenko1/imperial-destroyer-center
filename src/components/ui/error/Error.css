.error-container {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 2rem;
}

.error-content {
  text-align: center;
  background: white;
  padding: 2rem;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.error-icon {
  width: 48px;
  height: 48px;
  line-height: 48px;
  border-radius: 50%;
  background: #fee2e2;
  color: #dc2626;
  font-size: 24px;
  font-weight: bold;
  margin: 0 auto 1rem;
}

.error-title {
  color: #dc2626;
  margin: 0 0 0.5rem;
  font-size: 1.5rem;
}

.error-message {
  color: #666;
  margin: 0 0 1.5rem;
}

.error-retry {
  background: var(--primary-color);
  color: white;
  border: none;
  padding: 0.5rem 1rem;
  border-radius: 4px;
  cursor: pointer;
  transition: opacity 0.2s;
}

.error-retry:hover {
  opacity: 0.9;
}
