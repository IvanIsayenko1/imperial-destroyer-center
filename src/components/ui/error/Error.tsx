import React from "react";
import "./Error.css";

interface ErrorProps {
  message?: string;
  onRetry?: () => void;
}

export const Error: React.FC<ErrorProps> = ({
  message = "Something went wrong",
  onRetry,
}) => {
  return (
    <div className="error-container">
      <div className="error-content">
        <div className="error-icon">!</div>
        <h2 className="error-title">Error</h2>
        <p className="error-message">{message}</p>
        {onRetry && (
          <button onClick={onRetry} className="error-retry">
            Try Again
          </button>
        )}
      </div>
    </div>
  );
};
