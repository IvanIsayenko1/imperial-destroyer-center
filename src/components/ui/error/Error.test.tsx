import { render, screen, fireEvent } from "@testing-library/react";
import { Error } from "./Error";

describe("Error component", () => {
  it("should render default message", () => {
    render(<Error />);
    expect(screen.getByText("Something went wrong")).toBeInTheDocument();
  });

  it("should render custom message", () => {
    const message = "Custom error message";
    render(<Error message={message} />);
    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it("should render the retry button if onRetry is provided", () => {
    const onRetry = vi.fn();
    render(<Error onRetry={onRetry} />);
    const button = screen.getByText("Try Again");
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(onRetry).toHaveBeenCalledTimes(1);
  });

  it("should not render the retry button if onRetry is not provided", () => {
    render(<Error />);
    const button = screen.queryByText("Try Again");
    expect(button).toBeNull();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
  });
});
