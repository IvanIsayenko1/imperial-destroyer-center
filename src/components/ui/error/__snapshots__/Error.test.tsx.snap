// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="error-container"
  >
    <div
      class="error-content"
    >
      <div
        class="error-icon"
      >
        !
      </div>
      <h2
        class="error-title"
      >
        Error
      </h2>
      <p
        class="error-message"
      >
        Something went wrong
      </p>
    </div>
  </div>
</DocumentFragment>
`;
